src/catchup/simulation/TxSimApplyTransactionsWork.cpp

```diff
@@ -411,7 +411,11 @@ TxSimApplyTransactionsWork::scaleLedger(
     // First, update transaction source accounts
     LedgerTxn ltx(mApp.getLedgerTxnRoot());
     mutateTxSourceAccounts(newEnv, ltx, keys, partition);
-    mutateOperations(newEnv, ltx, keys, partition);
+
+    // Maps PoolID from ChangeTrustOp to the LiquidityPoolParameters so we can
+    // scale poolIds for pools created and deposited into in the same ledger
+    UnorderedMap<PoolID, LiquidityPoolParameters> ctPoolIdToParam;
```

Comment:
Seems like `mutateOperations` does not really need `ctPoolIdToParam`, as it can be tracked inside the function

Reply:
Also, loading LPs from the DB during scaling may have a pretty noticeable performance impact, so you may want to consider caching simulated LP mappings.

Reply:
Moved `ctPoolIdToParam` into `mutateOperations`. For your second point, wouldn't `mEntryCache` in `LedgerTxnRoot` cache the LP after the first query? Or am I misunderstanding?